_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp

```diff
@@ -2830,7 +2830,7 @@ void UpdateDPB(
     MFX_SORT_STRUCT((dpb + st0), mfxU16(end - st0), m_poc, >);
 
     // sliding window over STRs
-    if (end && end == par.mfx.NumRefFrame)
+    if (end && end == MAX_DPB_SIZE)
```

Comment:
You're breaking sliding window here.
HEVC encoder DPB should not contain more refs than NumRefFrame.
If NumRefFrame > MAX_DPB_SIZE consider it parameters validation issue.